src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1528,7 +1530,10 @@ tx_gateway_frontend::do_commit_tm_tx(
         outcome->set_value(tx_errc::unknown_server_error);
         throw;
     }
-    outcome->set_value(tx_errc::none);
+
+    if (!is_transaction_ga()) {
+        outcome->set_value(tx_errc::none);
+    }
```

Comment:
prior to this change `do_commit_tm_tx` looks like it always set a value or exception on `outcome`. but now it looks like right below this we might return without doing that.

can you address that concern?

Reply:
Yes, It is a bug. Thanks!